test/utils/image/manifest.go

```diff
@@ -0,0 +1,104 @@
+
+import (
+	"fmt"
+	"runtime"
+)
+
+const (
+	e2eRegistry     = "gcr.io/kubernetes-e2e-test-images"
+	gcRegistry      = "gcr.io/google-containers"
+	PrivateRegistry = "gcr.io/k8s-authenticated-test"
+)
+
+type ImageConfig struct {
+	registry string
+	name     string
+	version  string
+}
+
+func (i *ImageConfig) SetRegistry(registry string) {
+	i.registry = registry
+}
+
+func (i *ImageConfig) SetName(name string) {
+	i.name = name
+}
+
+func (i *ImageConfig) SetVersion(version string) {
+	i.version = version
+}
+
+var (
+	ClusterTester      = ImageConfig{e2eRegistry, "clusterapi-tester", "1.0"}
+	CudaVectorAdd      = ImageConfig{e2eRegistry, "cuda-vector-add", "1.0"}
+	Dnsutils           = ImageConfig{e2eRegistry, "dnsutils", "1.0"}
+	EntrypointTester   = ImageConfig{e2eRegistry, "entrypoint-tester", "1.0"}
+	Fakegitserver      = ImageConfig{e2eRegistry, "fakegitserver", "1.0"}
+	Frontend           = ImageConfig{e2eRegistry, "frontend", "1.0"}
+	Goproxy            = ImageConfig{e2eRegistry, "goproxy", "1.0"}
+	Hostexec           = ImageConfig{e2eRegistry, "hostexec", "1.0"}
+	Iperf              = ImageConfig{e2eRegistry, "iperf", "1.0"}
+	JessieDnsutils     = ImageConfig{e2eRegistry, "jessie-dnsutils", "1.0"}
+	Kitten             = ImageConfig{e2eRegistry, "kitten", "1.0"}
+	Liveness           = ImageConfig{e2eRegistry, "liveness", "1.0"}
+	LogsGenerator      = ImageConfig{e2eRegistry, "logs-generator", "1.0"}
+	Mounttest          = ImageConfig{e2eRegistry, "mounttest", "1.0"}
+	MounttestUser      = ImageConfig{e2eRegistry, "mounttest-user", "1.0"}
+	Nautilus           = ImageConfig{e2eRegistry, "nautilus", "1.0"}
+	Net                = ImageConfig{e2eRegistry, "net", "1.0"}
+	Netexec            = ImageConfig{e2eRegistry, "netexec", "1.0"}
+	Nettest            = ImageConfig{e2eRegistry, "nettest", "1.0"}
+	NginxSlim          = ImageConfig{gcRegistry, "nginx-slim", "0.20"}
+	NginxSlimNew       = ImageConfig{gcRegistry, "nginx-slim", "0.21"}
+	NoSnatTest         = ImageConfig{e2eRegistry, "no-snat-test", "1.0"}
+	NoSnatTestProxy    = ImageConfig{e2eRegistry, "no-snat-test-proxy", "1.0"}
+	NWayHTTP           = ImageConfig{e2eRegistry, "n-way-http", "1.0"}
+	Pause              = ImageConfig{gcRegistry, "pause", "3.0"}
+	Porter             = ImageConfig{e2eRegistry, "porter", "1.0"}
+	PortForwardTester  = ImageConfig{e2eRegistry, "port-forward-tester", "1.0"}
+	Redis              = ImageConfig{e2eRegistry, "redis", "1.0"}
+	Redisslave         = ImageConfig{e2eRegistry, "redis-slave", "1.0"}
+	ResourceConsumer   = ImageConfig{e2eRegistry, "resource-consumer", "1.1"}
+	ResourceController = ImageConfig{e2eRegistry, "resource-consumer/controller", "1.0"}
+	ServeHostname      = ImageConfig{e2eRegistry, "serve-hostname", "1.0"}
+	TestWebserver      = ImageConfig{e2eRegistry, "test-webserver", "1.0"}
+)
+
+func GetE2EImage(image ImageConfig) string {
+	return fmt.Sprintf("%s/%s-%s:%s", image.registry, image.name, runtime.GOARCH, image.version)
```

Comment:
this logic assumes that the host running tests matches the cluster's arch. we probably should instead be detecting the arch of the cluster, somehow.

this doesn't need to be fixed now, but it's worth keeping in mind.

Reply:
in fact, we've already done this twice before: https://github.com/kubernetes/kubernetes/blob/a64eeb47f6ed4aa2587113793a76878b876809f9/test/e2e/framework/util.go#L215-L234
https://github.com/kubernetes/kubernetes/blob/a64eeb47f6ed4aa2587113793a76878b876809f9/test/integration/framework/util.go#L35-L53

Reply:
sure, I'll take a look as a follow up PR to consolidate everything